src/wallet/walletdb.h

```diff
@@ -94,12 +99,18 @@ class CKeyMetadata
         READWRITE(this->nVersion);
         nVersion = this->nVersion;
         READWRITE(nCreateTime);
+        if (this->nVersion >= VERSION_WITH_HDDATA)
+        {
+            READWRITE(hdKeypath);
+            READWRITE(hdMasterKeyID);
+        }
     }
 
     void SetNull()
     {
         nVersion = CKeyMetadata::CURRENT_VERSION;
         nCreateTime = 0;
+        hdKeypath.clear();
```

Comment:
Good catch. Thanks.
Added a commit to address this: https://github.com/bitcoin/bitcoin/pull/8323/commits/68d7682b9f8c0f548e0a9ef03296320b8ae3960d